### ISSUE TITLE
[EPIC] Aging content MVP: Aging Content New Framework for Notifications

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#15504]
## Status:
**[2024-09-09]** Notifications turned on for the Full Width Banner Alert. Two additional stories (and a KB article update) need to be completed before this is considered finished. Updating the text on the notification emails, and testing/configuring the ability to send test emails to specific people for QA in lower environment before releasing to prod.
**[2024-07-08]** Work is almost completed, at which time notifications can be turned on. Per Michelle M, no review is needed; just notification to manager of editors.
**[2024-03-13]** Framework has been created; notifications/testing/documentation will be worked on in Sprint 106 & 107

**[01/09/2024] Outcome of Drupal refinement meeting today:**
cc @dsasser @Becapa @jilladams 
  * A Planned Architecture ticket needs to be created, with the Planned Architecture document being the first artifact created for the MVP of Aging Content. Daniel S will create the ticket, which will be pulled into the next sprint.
    * Q&A during the meeting:
      * Q - Michael: Are the timeframes per content type, or will they be more granular e.g. one full-width banner might have X timeframe, and another full-width banner might have Y timeframe. A: The timeframe will be at the content-type level, not granular.
      * Q - Daniel: Is it a hard rule that the timeframe needs to be in the code? For instance, can it be something the Admin sets up when they set up the node? A - Fran: No, but it is a hard rule that the editors can't control the timeframe/shouldn't see any way to control the timeframe. Either is fine.
  * Fran to engage Bernie X and Marcia O on the CMS team to give them a heads up about the work, and to ask if Edmund might have the time to review it/collaborate with Daniel on it.
  * When the document is ready, notify CMS Collab team to see if they want to do an async review, or a formal CMS Collab Review Cycle.

**[12/4/2023 Status]** 
* Discussed with Dave in 12/4 Product Sync meeting. Action Items for @FranECross > Create three Initiatives, (1) Content Types Notifications for Aging Content; (2) Blocks Notifications for Aging Content; and (3) Archiving Aged Content Types & Blocks

## Elevator Statement
Allow for content-type specific time frames for email reminders with unique messaging.

## Overview
Content in general, and for the purpose of this Initiative, content types of Full-width banners in VA.gov become stale and aren’t monitored or updated on a regular cadence, allowing many of them to still display after a campaign or event has passed. Full-width banners should be used only for informational alerts or warning alerts.

Currently, there is a Notification system set on a 365 day cadence from set from the banner's  Created date or if subsequently edited, Last Edited date.

To provide timely information to Veterans and other users of VA.gov, and ensure banners are replaced and/or monitored, we need to implement the ability for the notification system to send notification emails based on a cadence specific to particular content types, e.g. Full-Width banners will only display for seven (7) days, and notifications will be sent three days before the 7th day (telling them they need to edit to add a note that it is still needed, or archive the banner by the seventh day), then the morning of the 7th day (telling them they need to either edit to add a note that the banner is still needed, or archive). 

If possible, this should also accommodate Home Page Benefit Promo Blocks, and Home Page News Spotlight Blocks, which will be set at 30 day notification sequences (future stories/initiative to actually implement). Take note that if other banner types are created in the future, then this framework needs to be extendable to accommodate them as well.

If not possible, then another framework to accommodate blocks should be created in later endeavor.

---

## Headlines
**VA stakeholders** - 

**OCTO lead** - Dave Conlon

**Is this project tracked to an upcoming event or congressional deadline?** No

### Details

How might we ensure a better site-user experience by encouraging appropriate use of Full-width banners for time-sensitive or frequently refreshed content, and in limited quantity. 

### Discovery
* See Discovery [document](products/content/Aging Content /Discovery for Content Types and Blocks - Aging Content)

#### Affected users and stakeholders

* CMS Editors of:
  * Home page news
  * Benefit** Promo information
  * Homepage Top Pages link
  * Full Width Banners
* Veterans and other users of VA.gov's unauthenticated experience

**Examples of outdated Full Width Banners still in VA.gov as of 11/30/2023:**
* National Buddy Check Week: The campaign timeframe was October 16 to 20th
* How Veterans Can Register to Vote: The image is outdated and shows a book with Tax year 2020/2021
* Recognizing LGBTQ+ Veterans During Pride Month: Pride month was in June

### Hypothesis

* If the main goal is to have no stale content, even if the space is blank, then the hypothesis could be the following.
  * We believe that by providing notifications to editors alerting them that their content is expiring, they will be prompted to add new content, thus ensuring the Veteran and other users of VA.gov have timely, up-to-date information in website.

## In Scope
* Framework is created that is extendable to other content types
* Two notification emails are sent to the last published author based on the expiration date coded; 3 days before expiration date and morning of expiration date.

## Out of Scope
* Notifications for Blocks
* Auto-archiving content

## Acceptance Criteria for the Framework and Full-Width Banner Notifications

- [ ] **AC1: Drupal engineer is able to code an expiration date for any content type**
   * **GIVEN** a new framework for defining expiration dates for any content type is created
   * **WHEN** the engineer defines an expiration date for a particular content type
   * **THEN** the expiration date can be used to then set up a notification email to be sent to the author who last published the full-width banner

- [ ] **AC2: Editor receives their 1st notification email informing them they should manually archive or edit to extend the timeframe**
   * **GIVEN** AC1 is completed
   * **AND** it is 3 days before Day 7
   * **THEN** the 1st notification email is sent to the editor who created the banner alerting them they need to manually archive or edit to extend the timeframe. See the text for the email in specific implementation ticket.

- [ ] **AC3: Editor receives their 2nd notification email on day 7 telling them they need to manually archive, or edit the banner to extend the timeframe**
   * **GIVEN** AC1 is completed
   * **AND** is the mornign of Day 7 of the timeframe
   * **THEN** the 2nd notification email will be sent to the editor who created the banner alerting them they should manually archive, or edit to extend the timeframe. See the text for the email in the specific implementation ticket.

- [ ] **AC4: Editor edits an existing full-width banner when still within chosen date range, and the expiration date is 'reset'**
   * **GIVEN** AC1 is completed
   * **AND** it is within the 7 day timeframe
   * **WHEN** the editor edits the banner so that the 7 day timeframe starts anew
   * **AND** they can successfully edit and save
   * **AND**the expiration date changes to seven days from edit date
   * **AND** the two notification emails will be sent according to the new 7-day timeframe.

- [ ] Requires KB article update
- [ ] a11y review for the new text on the form, and the email body

- [ ] Requires KB article update
- [ ] a11y review for the new text on the form, and the email body

### Change management triage

The change represented by this user story will:
- [ ] Be a site-wide change to appearance or key functionality (such as log-in process)
- [ ] Be a specific change that requires more than 30 minutes of work by a user to meet a publishing requirement
- [ ] Remove a piece of functionality (such as restricting editing functions, or “hardening”)
- [ ] Require action by some or all users by a specific deadline
- [ ] Change the front end and be highly visible to the public
- [ ] Noticeable performance improvements (publishing speed, predictability)

If you selected an item above, use the [Change Management Runbook](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/facilities/change-management) to create a CM epic and associated tickets.

If you did not select an item above, update issues/PRs in GitHub but don’t plan for change management.

## Design principles

Veteran-centered
- [ ] `Single source of truth`: Increase reliability and consistency of content on VA.gov by providing a single source of truth.
- [ ] `Accessible, plain language`: Provide guardrails and guidelines to ensure content quality.
- [ ] `Purposely structured content`: Ensure Content API can deliver content whose meaning matches its structure.
- [X] `Content lifecycle governance`: Produce tools, processes and policies to maintain content quality throughout its lifecycle.

Editor-centered
- [X] `Purpose-driven`: Create an opportunity to involve the editor community in VA’s mission and content strategy goals.
- [ ] `Efficient`: Remove distractions and create clear, straightforward paths to get the job done.
- [ ] `Approachable`: Offer friendly guidance over authoritative instruction.
- [ ] `Consistent`: Reduce user’s mental load by allowing them to fall back on pattern recognition to complete tasks.
- [X] `Empowering`: Provide clear information to help editors make decisions about their work.

## Runbook
<optional>

**Comment by jilladams** [comment]
Today, Erika & Michelle talked to VHA DM about the fact that the notifications system emails the last editor of a piece of content. Sometimes, that's not the decision-making owner of the content. They are contemplating making it possible to set an owner or specific email address, to help manage that. They'll loop in @davidconlon as that convo develops.

**Comment by jilladams** [comment]
@swirtSJW FYSA in Q4, the PW team will be piling onto the notification email system to do some aging content notifications (stubbed out in this epic). We'll want to schedule some time with you and @chri5tia @dsasser for a high level step through of what you learned, how it works, powers and limitations. I'm suggesting we do this internal to Sitewide, since it's fresh on your mind, but let me know if you think it's wise to include CMS Team folks when we get there. (Our Q4 work for notifications includes 1 content type, 2 block types, and 1 Taxonomy, so we do know there will likely be some blocking work required to get the system working with not-content-types.)

**Comment by EricaRobbins** [comment]
@davidconlon - this seems to straddle both CMS and Public Websites, where does this belong?

**Comment by mmiddaugh** [comment]
**Update 6.3.2024**

- ECA (Events/Conditions/Actions) framework is built
- Intending to apply to full width banners as first use case

**Comment by davidconlon** [comment]
We completed the framework and have begun piloting with full-width banners and the framework can support many different permutations.

**Comment by FranECross** [comment]
Associated tickets added as sub-issues for viewing in new Gihub project board

**Comment by FranECross** [comment]
Closing as completed; moved the eca email ticket over to the new [ECA epic]([[EPIC] ECA Framework expansion to other types of actions](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18868)
#18868)
